### DISCUSSION TITLE
Custom NER training - understanding loss and learn rate schedule

### DISCUSSION THREAD

**Discussion by deepakdalakoti** [explosion/spaCy#10682] [Help: Model Advice] [answered]
Hi,

I am training a custom NER model. I am trying to understand the output of the logger. I can see that the model outputs a few metrics including LOSS NER, ENTS_F, ENTS_P, ENTS_R, however I am not sure what dataset these metrics are calculated on. From my understanding so far, LOSS NER is calculated on the training set while other metrics (ENTS_F, ENTS_P, ENTS_R) are calculated on the dev set. Is that true.

A second question around implementing learning rate scheduler. I can see thinc provides some registry functions which I can use to change my learning rate. I am however not sure if the learning rate is updated every step or every epoch. Any clarifications would be really appreciated. 

<img width="315" alt="Capture" src="https://user-images.githubusercontent.com/20105140/164355685-8173ae95-04ce-4495-ad99-d35a4e05450a.PNG">

**Comment by thomashacker** [comment, marked as answer]
Hello, 
You are correct. The performance metrics (`ENTS_F`, `ENTS_P`, `ENTS_R`) are calculated on the development set and the loss on the training set. The learning rate schedules for the `optimizer` are [updated every step](https://github.com/explosion/spaCy/blob/master/spacy/training/loop.py#L220).

**Reply by yaelkurz** [reply]
what does Loss transformer and Loss NER refer to?

**Reply by thomashacker** [reply]
The loss indicates how well the model generalizes the `training` data and is used to perform backpropagation. The performance metrics are evaluated on the `development` data.
